File: nhglib-core/src/io/github/movementspeed/nhglib/utils/scenes/SceneMappings.java
package io.github.movementspeed.nhglib.utils.scenes;

import io.github.movementspeed.nhglib.data.models.serialization.ComponentJson;

import java.util.HashMap;

/**
 * Created by Fausto Napoli on 20/12/2016.
 */
public class SceneMappings {
    private static HashMap<String, Class<?>> assetClassesMapping = new HashMap<>();
    private static HashMap<String, Class<? extends ComponentJson>> componentJsonClassesMapping = new HashMap<>();

    public static void addAssetClassMapping(String type, Class<?> assetClass) {
        assetClassesMapping.put(type, assetClass);
    }

    public static void addComponentJsonMapping(String type, Class<? extends ComponentJson> componentClass) {
        componentJsonClassesMapping.put(type, componentClass);
    }

    public static Class assetClassFromAlias(String alias) {
        return assetClassesMapping.get(alias);
    }

    public static ComponentJson componentJsonFromType(String type) {
        ComponentJson componentJson = null;

        try {
            Class componentJsonClass = componentJsonClassFromType(type);
            componentJson = (ComponentJson) componentJsonClass.newInstance();
        } catch (InstantiationException | IllegalAccessException e) {
            e.printStackTrace();
        }

        return componentJson;
    }

    private static Class<? extends ComponentJson> componentJsonClassFromType(String type) {
        return componentJsonClassesMapping.get(type);
    }
}

File: nhglib-core/src/io/github/movementspeed/nhglib/utils/scenes/SceneMappings.kt
package io.github.movementspeed.nhglib.utils.scenes

import io.github.movementspeed.nhglib.data.models.serialization.ComponentJson

import kotlin.collections.HashMap
import kotlin.reflect.KClass

/**
 * Created by Fausto Napoli on 20/12/2016.
 */
object SceneMappings {
    private val assetClassesMapping = HashMap<String, KClass<*>>()
    private val componentJsonClassesMapping = HashMap<String, KClass<out ComponentJson>>()

    fun addAssetClassMapping(type: String, assetClass: KClass<*>) {
        assetClassesMapping[type] = assetClass
    }

    fun addComponentJsonMapping(type: String, componentClass: KClass<out ComponentJson>) {
        componentJsonClassesMapping[type] = componentClass
    }

    fun assetClassFromAlias(alias: String): KClass<*>? {
        return assetClassesMapping[alias]
    }

    fun componentJsonFromType(type: String): ComponentJson? {
        var componentJson: ComponentJson? = null

        try {
            val componentJsonClass = componentJsonClassFromType(type)
            componentJson = componentJsonClass?.objectInstance
        } catch (e: InstantiationException) {
            e.printStackTrace()
        } catch (e: IllegalAccessException) {
            e.printStackTrace()
        }

        return componentJson
    }

    private fun componentJsonClassFromType(type: String): KClass<out ComponentJson>? {
        return componentJsonClassesMapping[type]
    }
}
